tests/test_ir.py

```diff
@@ -286,3 +286,24 @@ def test_descriptions():
 \t  %0 : !i32 = arith.constant() ["value" = 1 : !i32]
 \t}
 )'''
+
+
+def test_op_custom_verify_is_done_last():
+    a = Constant.from_int_and_width(1, i32)
```

Comment:
I'd recommend a different test, maybe something like this:
``` python
@irdl_op_def
class CustomVerifyLastOp(Operation):
    name = 'test.custom_verify_last_op'
    val: Annotated[Operand, i64]

    @staticmethod
    def get(val: SSAValue):
        return CustomVerifyLastOp.build(operands: [val])

    def verify_(self):
        raise VerificationError('should not be triggered')

def test_op_custom_verify_is_done_last():
    a = Constant.from_int_and_width(1, i32)
    b = CustomVerifyLastOp.get(a.res)
    with pytest.raises(VerificationError) as e:
        b.verify
    assert e.value.args[0] == 'I forget what the message is supposed to be here'
```

Reply:
Agreed!

Reply:
The janky system is also coupled to the implementation. At the end of the day, we're testing irdl, so it feels like representative code that a user might write is best.